File: Baekjoon/C++/4796. 캠핑/solution.cpp
// https://www.acmicpc.net/problem/4796

#include <algorithm>
#include <iostream>

int main(int argc, char const *argv[])
{
    std::cin.tie(0);
    std::cout.tie(0);

    int L, P, V, q, r, index = 1;
    while (true)
    {
        std::cin >> L >> P >> V;
        if (L == 0 && P == 0 && V == 0)
        {
            break;
        }
        // (휴가에 들어갈수있는 연속일수 * 연속일수중 캠핑장을 사용가능한 일수) + 남은 일수중 최대한 캠핑장을 사용가능한 일수를 넣음
        q = V / P;
        r = V % P;
        std::cout << "Case " << index << ": " << q * L + std::min(r, L) << "\n";
        index += 1;
    }
    return 0;
}

File: Baekjoon/Python/4796. 캠핑/solution.py
# https://www.acmicpc.net/problem/4796

import sys

input = sys.stdin.readline

index = 1
while True:
    L, P, V = map(int, input().split())
    if L == P == V == 0:
        break
    total = 0
    q, r = divmod(V, P)
    # (휴가에 들어갈수있는 연속일수 * 연속일수중 캠핑장을 사용가능한 일수) + 남은 일수중 최대한 캠핑장을 사용가능한 일수를 넣음
    total += q * L + min(r, L)
    print(f"Case {index}: {total}")
    index += 1
